usr/src/uts/common/io/drm/drm_sunmod.c

```diff
@@ -210,23 +213,49 @@ drm_gem_map(devmap_cookie_t dhp, dev_t dev, uint_t flags, offset_t off,
 	if (!minor->dev)
 		return (ENODEV);
 
-	drm_dev = minor->dev;
+	dev = minor->dev;
 
-	mutex_enter(&drm_dev->struct_mutex);
-	map = __find_local_map(drm_dev, off);
+	mutex_enter(&dev->struct_mutex);
+	map = __find_local_map(dev, off);
 	if (!map) {
-		mutex_exit(&drm_dev->struct_mutex);
+		mutex_exit(&dev->struct_mutex);
 		*pvtp = NULL;
 		return (DDI_EINVAL);
 	}
 
-	mutex_exit(&drm_dev->struct_mutex);
+	dhp = (devmap_handle_t *)dhc;
+	cp = (struct ddi_umem_cookie *)dhp->dh_cookie;
+#ifdef DEBUG
+	if (cp->cook_refcnt != 0)
+		DRM_DEBUG_DRIVER("cookie is not zero: %d\n", cp->cook_refcnt);
+#endif
+	cp->cook_refcnt = 1;
+
+	mutex_exit(&dev->struct_mutex);
```

Comment:
Seems like given the code at 232, you're really wanting to assert this, or you'll be leaking refs?

Either that, or you should increment at 232